File: quarto_semestre/POO/tdeaula-12/src/main/java/com/poo/tde/rules/shared/Entity.java
package com.poo.tde.rules.shared;

import com.poo.tde.rules.shared.valueobjects.Id;

public class Entity {
    protected Id id;
    public Entity() {
        id = new Id();
    }

    public boolean equals(Entity entity) {
        return this.id.getValue() == entity.id.getValue();
    }

    public boolean different(Entity entity) {
        return this.equals(entity);
    }
}

File: quinto-semestre/modelagem-de-software/aula-1/src/rules/shared/entity.ts
import Id from "./value-objects/Id";

export default class Entity {
    readonly id: Id;

    constructor(
        id: string
    ) {
        this.id = new Id(id);
    }

    equals(entity: Entity): boolean {
        return this.id.value === entity.id.value;
    }

    different(entity: Entity): boolean {
        return this.equals(entity);
    }
}